hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/Listing.java

```diff
@@ -464,9 +466,10 @@ private boolean buildNextStatusBatch(S3ListResult objects) {
         // Skip over keys that are ourselves and old S3N _$folder$ files
         if (acceptor.accept(keyPath, s3Object) && filter.accept(keyPath)) {
           S3AFileStatus status = createFileStatus(keyPath, s3Object,
-                  listingOperationCallbacks.getDefaultBlockSize(keyPath),
-                  getStoreContext().getUsername(),
-                  s3Object.eTag(), null, isCSEEnabled);
+              listingOperationCallbacks.getDefaultBlockSize(keyPath),
```

Comment:
let move username and blocksize retrievel out of the for() loop...they don't change after all

Reply:
ack